### PR TITLE
chore(package): update `defaults 1.4.0` support Webpack@3

### PR DESCRIPTION
**Pull request by mattlewis92**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ matrix:
   fast_finish: true
```
Comment (michael-ciniawsky) [inline .travis.yml]: Could you revert the `travis.yml` changes please, we have a organsation wide upgrade solution for this in place :)

**Comment by mattlewis92** [comment]
@bebraw done (on a side note `webpack-defaults` is an awesome idea for keeping multiple repos up to date with the same scaffolding! 👍 )

File: package.json [hunk 2]
```diff
@@ -3,29 +3,29 @@
   "version": "4.0.4",
   "author": "Johannes Ewald @jhnns",
   "description": "Less loader for webpack. Compiles Less to CSS.",
-  "main": "dist/index.js",
+  "main": "dist/cjs.js",
   "scripts": {
     "create-spec": "babel-node test/helpers/createSpec.js",
     "pretest": "npm run create-spec",
```
Comment (michael-ciniawsky) [inline package.json]: The `create-spec` (`pretest`) `script` doesn't seem to run on the `PART=coverage` build https://travis-ci.org/webpack-contrib/less-loader/jobs/244975479

Comment (mattlewis92) [inline package.json]: Ah I was wondering why that was failing, should be fixed now! 😄

**Comment by mattlewis92** [comment]
Got the appveyor tests passing (had to remove the quote marks added to commands + use npm 5). I also removed the `cjs.js` file as this lib doesn't actually use import + export syntax so it would have failed when published as no `module.exports.default` would have been created. I think this should be good to go now.

**Comment by bebraw** [comment]
Removing `cjs.js` might not be the right move as **webpack-defaults** will regenerate it on the next run. Instead I would port `index.js` to ES6 style import/export to live with it.

**Comment by mattlewis92** [comment]
Done. It might be worth adding https://github.com/59naga/babel-plugin-add-module-exports to webpack-defaults for a cleaner solution.

**Comment by bebraw** [comment]
Thanks for the fixes. Let's wait for some feedback.

I have a feeling we might have discussed about **babel-plugin-add-module-exports** in the past and decided against it. But thanks for the idea. 👍

**Comment by d3viant0ne** [comment]
@jhnns Has already added defaults to `less-loader` though it needs to be updated. The loader runs on webpack-defaults standard. 

@mattlewis92 - If you think there is something Babel(wise) that would be beneficial, open up an issue in `webpack-contrib/webpack-defaults` and we can discuss it there at an organization level.

File: appveyor.yml [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
+    - nodejs_version: '8'
+      webpack_version: 3.0.0
       job_part: test
-    - nodejs_version: 6
+    - nodejs_version: '6'
+      webpack_version: 3.0.0
       job_part: test
-    - nodejs_version: 4
+    - nodejs_version: '4.3'
+      webpack_version: 3.0.0
```
Comment (d3viant0ne) [inline appveyor.yml]: Flip the version to `latest` from `3.0.0` 

&& 

change `cmd: npm install webpack@^%webpack_version%`

to 

`cmd: npm install webpack@%webpack_version%`

This needs to be updated in defaults

File: package.json [hunk 4]
```diff
@@ -38,7 +39,7 @@
   },
   "peerDependencies": {
     "less": "^2.3.1",
-    "webpack": "^2.2.0"
+    "webpack": "^2.0.0 || >= 3.0.0-rc.0 || ^3.0.0"
```
Comment (d3viant0ne) [inline package.json]: Pull the RC here, we don't want to officially support pre-release builds now that final is out on npm

`"webpack": "^2.0.0 || ^3.0.0"`

File: package.json [hunk 5]
```diff
@@ -59,8 +60,8 @@
     "nsp": "^2.6.3",
     "pre-commit": "^1.2.2",
     "standard-version": "^4.0.0",
-    "webpack": "^2.3.2",
-    "webpack-defaults": "^1.0.1"
+    "webpack": "^3.0.0",
```
Comment (d3viant0ne) [inline package.json]: Remove this and add it as a `peerDependency`. The CI configs install the proper versions of webpack & developers can install if & when needed.

**Comment by d3viant0ne** [comment]
For reference, if it touches source code in any way ( to include `import` ) it is not a chore.

File: package.json (hunk 5 above)
Comment (mattlewis92) [inline package.json]: I don't think this is the right action to take - it increases friction for a new person checking out the repo and just wanting to `npm install && npm test`. Secondly the default for npm install in v5 is to always save to package.json and package-lock.json - so just doing `npm install webpack` to develop locally is always going to update those files unless the user explicitly adds the `--no-save` flag which again causes confusion and decreases the likliehood of someone wanting to contribute (and will probably lead to a user accidentally committing the changes to a PR and having to have this discussion again).

**Comment by mattlewis92** [comment]
@d3viant0ne I've made the requested changes in appveyor.yml but left off removing the webpack dev dependency for now. I saw the discussion about the module exports babel plugin in https://github.com/webpack-contrib/webpack-defaults/issues/23 so I won't bother making an issue there as I can see it's already been discussed 😄

File: package.json (hunk 5 above)
Comment (d3viant0ne) [inline package.json]: Standard for the organization & not optional.

Comment (mattlewis92) [inline package.json]: Fair enough, should be fixed now.

**Comment by realpunk** [comment]
👍 waiting for merge :) ⏳

File: package.json [hunk 6]
```diff
@@ -3,29 +3,30 @@
...
     "clean:dist": "del-cli dist",
     "lint-staged": "lint-staged",
-    "prebuild": "yarn run clean",
-    "prepublish": "yarn run build",
-    "release": "yarn run standard-version",
+    "prebuild": "npm run clean",
+    "prepublish": "npm run build",
+    "release": "standard-version",
     "security": "nsp check",
     "serve:dev": "nodemon $2 --exec babel-node",
```
Comment (michael-ciniawsky) [inline package.json]: Can be removed :)

File: package.json [hunk 7]
```diff
@@ -3,29 +3,30 @@
...
-    "travis:test": "yarn run test",
-    "appveyor:test": "npm test",
+    "travis:test": "npm run test -- --runInBand",
+    "appveyor:test": "npm run test",
     "lint": "eslint --cache src test",
     "webpack-defaults": "webpack-defaults",
-    "start": "yarn run build -- -w",
+    "start": "npm run build -- -w",
     "build": "cross-env NODE_ENV=production babel src -d dist --ignore 'src/**/*.test.js'",
     "clean:dist": "del-cli dist",
```
Comment (michael-ciniawsky) [inline package.json]: Can be removed :), bc of https://github.com/webpack-contrib/less-loader/pull/208/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R30

File: package.json (hunk 5 above)
Comment (d3viant0ne) [inline package.json]: @mattlewis92 - FYI webpack-contrib/webpack-defaults#78

**Comment by mattlewis92** [comment]
I've updated webpack-defaults to pull in the latest changes from there. There is a bug with npm 5.0.x though that was making the CI fail so I had to force appveyor and travis to use npm 5.1.0, once the next node 8.x is released with npm 5.1.0 included these 2 hacks can be removed (by running `npm run webpack-defaults`)

**Comment by d3viant0ne** [comment]
@mattlewis92 - I'll fix the npm version issue upstream.

**Pull request merged**